### PR TITLE
Downgrade abseil-cpp

### PR DESCRIPTION
**Pull request by fmeum**
Also add a test that repeatedly starts a Bazel client and verifies that it reaches the server.

Work towards #22692

### CODE REVIEW

File: src/test/py/bazel/bazel_server_mode_test.py [hunk 1]
```diff
@@ -19,14 +19,24 @@
...
-    self.ScratchFile('WORKSPACE')
+    self.ScratchFile('MODULE.bazel')
 
     _, stdout, _ = self.RunBazel(['info', 'server_pid'])
     pid1 = stdout[0]
     _, stdout, _ = self.RunBazel(['info', 'server_pid'])
     pid2 = stdout[0]
     self.assertEqual(pid1, pid2)
 
+  def testBazelServerModeStressTest(self):
```
Comment (fmeum) [inline src/test/py/bazel/bazel_server_mode_test.py:30]: I can't reproduce the previous failure with this test on my local Linux machine.

**Comment by fmeum** [comment]
@meteorcloudy The build is failing because a patch fails to apply. I don't understand what's going on here, didn't this release work in the past?

**Comment by meteorcloudy** [comment]
The failure is due to https://github.com/bazelbuild/bazel/commit/193b114287b3e20850a4b106b889771dfa63a601, can you please remove the override but leave the patch file in third_party?

File: src/test/py/bazel/bazel_server_mode_test.py (hunk 1 above)
Comment (meteorcloudy) [reply src/test/py/bazel/bazel_server_mode_test.py:30]: Hmm, looks like even the existing test case should have caught this.

**Comment by fmeum** [comment]
Done

**Review by meteorcloudy — APPROVED** [review]

File: src/test/py/bazel/bazel_server_mode_test.py (hunk 1 above)
Comment (meteorcloudy) [reply src/test/py/bazel/bazel_server_mode_test.py:30]: I'll verify and fix this later if necessary.

Comment (meteorcloudy) [reply src/test/py/bazel/bazel_server_mode_test.py:30]: testBazelServerModeStressTest did catch the failure on my linux workstation. The first test case only somethings fails. But anyway this PR should be good enough, thanks!

**Pull request closed without merging**